grai-server/app/workspaces/models.py

```diff
@@ -0,0 +1,81 @@
+import uuid
+
+from django.db import models
+from rest_framework_api_key.models import AbstractAPIKey, BaseAPIKeyManager
+from django_multitenant.models import TenantModel
+
+
+class Workspace(TenantModel):
+    tenant_id = "id"
+
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    name = models.CharField(max_length=255)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        indexes = [
+            models.Index(fields=["name"]),
+        ]
+
+
+class Membership(models.Model):
```

Comment:
Is `Membership` synonymous with a role in this case? The role is potentially customizable in the future but it defines the access control rules within the workspace?

Reply:
Indeed, each user can have one membership to link them to a workspace.  There is then a text role field.